src/framework/control.ml

```diff
@@ -577,7 +581,10 @@ struct
     if get_bool "exp.cfgdot" then
       CfgTools.dead_code_cfg file (module Cfg : CfgBidir) liveness;
 
-    Spec.finalize ();
+    let marshal = Spec.finalize () in
+    if get_string "save_run" <> "" then (
+      Serialize.marshal marshal (Filename.concat (get_string "save_run") "spec_marshal")
+    );
```

Comment:
`Spec.finalize` is much later than the rest of marshaling, not sure if this matters.

Reply:
When marshaling happens here is irrelevant.

Reply:
If marshaling is to be done as a single object, like I'm now thinking above, this cannot be separate like this, but everything is unmarshaled/marshaled together. This just means all the solver data etc needs to be kept around for saving this late and all the solver etc data needs to be loaded earlier for `init`.